tools/serverpod_cli/lib/src/generator/types_record.dart

```
@@ -0,0 +1,116 @@
+    return null;
+  }
+
+  var (fieldsString, nullable) = _unwrapRecord(trimmedRecordInput);
+
+  var splitFields = splitIgnoringBracketsAndBracesAndQuotes(
+    fieldsString,
+    returnEmptyParts: true,
+  );
+
+  // Field into split from string must have at least 2 parts,
+  // meaning the record is using a `,` even if it has only 1 fields, e.g. `(int,)`
+  // or some named parameter like `({String foo})`.
+  if ((splitFields.length < 2 || splitFields.every((f) => f.isEmpty)) &&
+      !(splitFields.length == 1 &&
+          splitFields.single.startsWith('{') &&
+          splitFields.single.endsWith('}'))) {
+    return null;
+  }
+
+  var recordFields =
+      splitFields.where((s) => s.isNotEmpty).expand((splitField) {
+    if (splitField.startsWith('{') && splitField.endsWith('}')) {
+      return _parseNamedRecordFields(
+        splitField,
+        extraClasses: extraClasses,
+      );
+    }
+
+    // could be either just a positional type, or a named positional type (like `int` or `int someNumber`, or even `Set<String> someSet`)
+    var parts = splitIgnoringBracketsAndBracesAndQuotes(
+      splitField,
+      separator: ' ',
+    );
+
+    if (parts.length > 1 && !parts.last.startsWith('<')) {
+      // if the last part is a name (and not a generic parameter), then we need to drop that
+      splitField = parts.take(parts.length - 1).join();
+    }
+
+    return [
+      parseType(
+        splitField,
+        extraClasses: extraClasses,
+      ),
+    ];
+  }).toList();
+
+  return TypeDefinition(
+    className: RecordKeyword.className,
+    generics: recordFields,
+    nullable: nullable,
+  );
+}
+
+Iterable<TypeDefinition> _parseNamedRecordFields(
+  String namedRecordFieldsPart, {
+  List<TypeDefinition>? extraClasses,
+}) sync* {
+  if (!namedRecordFieldsPart.startsWith('{') ||
+      !namedRecordFieldsPart.endsWith('}')) {
+    throw Exception(
+      '"$namedRecordFieldsPart" does not described the named parameters of a Record',
+    );
+  }
```

Suggestion: Remove this.

As far as I can tell this is already validated on the only callsite.

Are there more synergies that can be simplified now that this is an implementation detail of `tryParseRecord`?

---

This could actually be converetd to a `_tryParseNamedRecordFields` that returns null if it does not describe the named parameters of a Record.

Then we only need to have this check in a single place and we don't need any implicit information.

---

(And the existance of these could be validated using the `start` and `end` variables.)

---

> (And the existance of these could be validated using the start and end variables.)

That's the thing, if the `{}` do not exist, the substring parsing would fail.

So what we could do then is `try/catch` everything, and return `null` for the entire record (I would not silently skip over any sub-parts which turn out not to be valid record (parts)).